test/dstu2_test.js

```diff
@@ -2,6 +2,25 @@ const cql = require('cql-execution');
 const cqlfhir = require('../src/index');
 const {expect} = require('chai');
 
+describe('#FHIRWrapper_DSTU2', () => {
+  let fhirWrapper;
+  let conditionResource;
+  before(() => {
+    fhirWrapper = cqlfhir.FHIRWrapper.FHIRv102();
+    conditionResource = require('./fixtures/dstu2/Condition_f201.json');
+  });
+
+  it('should wrap a fhir resource to the correct type when type not specified', () => {
+    let fhirObject = fhirWrapper.wrap(conditionResource);
+    expect(fhirObject.getTypeInfo().name).to.equal('FHIR.Condition');
+  });
+
+  it('should wrap a fhir resource to the type specified', () => {
+    let fhirObject = fhirWrapper.wrap(conditionResource, 'Observation');
+    expect(fhirObject.getTypeInfo().name).to.equal('FHIR.Observation');
```

Comment:
This behavior surprised me a little.  I wonder if it should throw if the JSON declares a `resourceType` but the caller tries to force it into an incompatible type.  It seems to me that whenever this happens, it's probably a mistake.

That said, I'm not sure if we can just do a straight string match -- because there is type inheritance.  For example, it's probably valid for a caller to pass in JSON with `"resourceType": "Condition"` but to ask for `'DomainResource'` (which is a parent type of "Condition").  Similarly, we might want to allow it the other way too (passed in `"resourceType": "DomainResource"` but asked for `'Condition'`) -- although we can't guarantee it's right, we also can't guarantee it's wrong and should give the caller the benefit of the doubt...  

Of course, to do this type of type-checking, you'd need to consult the model-info, which does bring in some additional complexity.  What do you think?

Reply:
I agree, I've added type checking which allows upcasting and downcasting but throws otherwise.